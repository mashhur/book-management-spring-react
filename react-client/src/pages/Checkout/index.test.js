import { shallow } from "enzyme";
import React from "react";
import { findByTestAttr } from "../../tests/testUtils";
import CheckOutWrapper from "./";

const setup = () => {
	return shallow(<CheckOutWrapper />);
};

describe("testing check out", () => {
	it("should render without any errors", async () => {
		const wrapper = setup();
		const component = await findByTestAttr(
			wrapper,
			"check-out-wrapper"
		);
		expect(component.length).toBe(1);
	});

	it("should load the correct snapshot", () => {
		const wrapper = setup();
		expect(wrapper).toMatchSnapshot();
	});

	it("should render AddressStep default", () => {
		const wrapper = setup();
		const component = wrapper.find("AddressStep");
		expect(component.length).toBe(1);
	});

	it("should render PaymentStep on state change", () => {
		React.useState = jest.fn(() => [1, jest.fn()]);
		const wrapper = setup();
		const component = wrapper.find("PaymentStep");
		expect(component.length).toBe(1);
	});
});
