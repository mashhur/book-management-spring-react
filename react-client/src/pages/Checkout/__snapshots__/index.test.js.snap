// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing check out should load the correct snapshot 1`] = `ShallowWrapper {}`;
